web/client/components/map/cesium/__tests__/Map-test.jsx

```diff
@@ -392,10 +392,12 @@ describe('CesiumMap', () => {
                 expect(dataSource.entities.values.length).toBe(4);
                 const mapCanvas = ref.map.canvas;
                 const { width, height } = mapCanvas.getBoundingClientRect();
-                simulateClick(mapCanvas, {
-                    clientX: width / 2,
-                    clientY: height / 2
-                });
+                setTimeout(() => { // this mitigates #9965 waiting for a better solution
```

Comment:
I noticed that the `ref.map.dataSourceDisplay.ready` inside the waitFor changes between state true/false twice. I would like to propose an additional change by switching the waitFor statement by checking the display ready before the entities length. I think we should keep also the setTimeout as you are proposing to have a buffer time.

```js
let prevReady = false;
let countReady = 0;
// the data source switches twice from false to true
// here we are waiting the second render
const checkReadyDataSource = () => {
    const currentReady = ref.map.dataSourceDisplay.ready;
    if (currentReady !== prevReady) {
        if (currentReady) {
            countReady += 1;
        }
        prevReady = ref.map.dataSourceDisplay.ready;
    }
    if (countReady === 2) {
        return true;
    }
    return false;
};
// first we check we got data source ready twice
// then we verify that the dataSources entities are available
waitFor(() => expect(checkReadyDataSource()
&& !!ref.map.dataSources.get(0).entities.values.length).toBe(true), {
    timeout: 5000
})
    .then(() => {
        expect(ref.map.dataSources.length).toBe(1);
        const dataSource = ref.map.dataSources.get(0);
        expect(dataSource).toBeTruthy();
        expect(dataSource.entities.values.length).toBe(4);
        const mapCanvas = ref.map.canvas;
        const { width, height } = mapCanvas.getBoundingClientRect();
        // adding additional timeout to ensure the complete render
        setTimeout(() => {
            simulateClick(mapCanvas, {
                clientX: width / 2,
                clientY: height / 2
            });
        }, 1000);
    })
    .catch(done);
```

Reply:
I tried to apply your suggestion inline.